### PR TITLE
executor: only reserve the sign bit when auto_random column is signed

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Problem Summary:

Currently, the highest bit (sign bit) of the `auto_random` column is reserved. However, in the case of unsigned column types, there is no sign bit, the highest bit can be used for sharding safely.

### What is changed and how it works?
When `CalcShard()` is called, the caller can determine whether to reserve the sign bit.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 
- The sign bit of `auto_random` column is reserved to prevent sharding only when the column type is signed.

### CODE REVIEW

File: meta/autoid/autoid.go [hunk 1]
```diff
@@ -890,3 +892,41 @@ func EncodeIntToCmpUint(v int64) uint64 {
 func DecodeCmpUintToInt(u uint64) int64 {
 	return int64(u ^ signMask)
 }
+
+// AutoRandomIDLayout is used to calculate the bits length of different section in auto_random id.
+// Layout(64 bits):
+// [zero_padding] [sign_bit] [shard_bits] [incremental_bits]
```
Comment (AilinKid) [inline meta/autoid/autoid.go:898]: Do we have lead-padding zero in auto_random now?

File: ddl/table.go [hunk 2]
```diff
@@ -586,7 +586,7 @@ func verifyNoOverflowShardBits(s *sessionPool, tbl table.Table, shardRowIDBits u
 	if err != nil {
 		return errors.Trace(err)
 	}
-	if tables.OverflowShardBits(autoIncID, shardRowIDBits, autoid.RowIDBitLength) {
+	if tables.OverflowShardBits(autoIncID, shardRowIDBits, autoid.RowIDBitLength, true) {
```
Comment (AilinKid) [inline ddl/table.go:589]: should judge the column signed character in the upper function caller and pass the parameter？

File: meta/autoid/autoid.go (hunk 1 above)
Comment (tangenta) [reply meta/autoid/autoid.go:898]: Yep. If the column type is not `bigint`, there is zero-padding in 64-bits layout.

File: ddl/table.go (hunk 2 above)
Comment (tangenta) [reply ddl/table.go:610]: `shard_rowid_bits` always reserve the sign bit.

**Pull request merged**